dds/DCPS/RTPS/RtpsDiscovery.cpp

```diff
@@ -125,6 +125,10 @@ RtpsDiscovery::Config::discovery_config(ACE_Configuration_Heap& cf)
       // spdpaddr defaults to DCPSDefaultAddress if set
       if (TheServiceParticipant->default_address() != ACE_INET_Addr()) {
         config->spdp_local_address(TheServiceParticipant->default_address());
+        ACE_TCHAR buff[256];
+        TheServiceParticipant->default_address().addr_to_string(&(buff[0]), 256);
+        OPENDDS_STRING addr_str(ACE_TEXT_ALWAYS_CHAR(&(buff[0])));
```

Comment:
```suggestion
        TheServiceParticipant->default_address().addr_to_string(buf, SOME_NAMED_CONSTANT);
        const OPENDDS_STRING addr_str(ACE_TEXT_ALWAYS_CHAR(buf));
```

Reply:
I avoided using buff directly because VS2019 was giving me a bounds warnings about array / pointer decay. I'll just put the cast in to keep it quiet.